app/controllers/kaui/queues_controller.rb

```diff
@@ -1,18 +1,45 @@
 class Kaui::QueuesController < Kaui::EngineController
-
   def index
     @account_id = params[:account_id]
+
+    unless params[:max_date].blank?
+      begin
+        max_date_test = Time.parse(params[:max_date]).iso8601
+      rescue StandardError
+      ensure
+        if max_date_test.nil?
+          flash[:error] = I18n.translate('errors.messages.invalid_max_date')
+          redirect_to account_path(@account.account_id) and return
+        end
+      end
+    end
+
+    unless params[:min_date].blank?
+      begin
+        min_date_test = Time.parse(params[:min_date]).iso8601
+      rescue StandardError
+      ensure
+        if min_date_test.nil?
+          flash[:error] = I18n.translate('errors.messages.invalid_min_date')
+          redirect_to account_path(@account.account_id) and return
+        end
+      end
+    end
+
     begin
       @now = Kaui::Admin.get_clock(nil, options_for_klient)['currentUtcTime'].to_datetime
     rescue KillBillClient::API::NotFound
       # If TestResource is not bound, then clock has not been manipulated, we can default to NOW
-      @now = DateTime.now.in_time_zone("UTC")
+      @now = DateTime.now.in_time_zone('UTC')
     end
 
-    min_date = params[:min_date] || '1970-01-01'
-    max_date = params[:max_date] || '2100-01-01'
+    min_date = Time.parse(params[:min_date]).iso8601 unless params[:min_date].blank? || '1970-01-01'
+    max_date = Time.parse(params[:max_date]).iso8601 unless params[:min_date].blank? || '2019-10-12T03:00:00+05:30'
```

Comment:
Should be `params[: max_date]`.

Also, maybe we should just default to `now` for the max date?

Reply:
> Also, maybe we should just default to `now` for the max date?

PR updated.

Reply:
Sorry for typo

Reply:
Added tests and url fix